### ISSUE TITLE
React generator generates wrong view code

### ISSUE THREAD

**Issue by antonioortizpola** [jhipster/generator-jhipster#8075]
##### **JHipster Version(s)**

```
wong@0.0.0 /home/*****/Documents/wong-app
└── generator-jhipster@5.1.0 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mibejoto.wong",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.1.0",
    "applicationType": "monolith",
    "baseName": "wong",
    "packageName": "com.mibejoto.wong",
    "packageFolder": "com/mibejoto/wong",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "yarn",
    "testFrameworks": [
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "es"
    ]
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity VehicleBrand (vehicle_brand) {
  name String required minlength(2)
}
entity VehicleBrandImage (vehicle_brand_image) {
  path String required
}
entity VehicleModel (vehicle_model) {
  name String required minlength(2)
}
entity VehicleVersion (vehicle_version) {
  name String required minlength(2)
}
entity VehicleVersionYear (vehicle_version_year) {
  year Integer required min(1950) max(2999)
}
relationship OneToMany {
  VehicleBrand{vehicleImage} to VehicleBrandImage{vehicleBrand},
  VehicleBrand{vehicleModel} to VehicleModel{vehicleBrand(name)},
  VehicleModel{vehicleVersion} to VehicleVersion{vehicleModel(name)},
  VehicleVersion{vehicleVersionYear} to VehicleVersionYear{vehicleVersion(name)}
}

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

git version 2.17.1

node: v8.11.3

npm: 5.6.0

yarn: 1.9.4

Docker version 18.06.0-ce, build 0ffa825

docker-compose version 1.21.2, build a133471

##### **Overview of the issue**

When you create some Jhipster entities:

- If you select something else than the id for the `display` property, this property is used not only for the HTML property of the option in the select, it also changes the value, generating an error when you try to save a new one, because the dependent object does not have the id.
Code created by default:
```
<AvInput
  id="vehicle-version-vehicleModel"
  type="select"
  className="form-control"
  name="vehicleModel.name"
  onChange={this.vehicleModelUpdate}
>
  <option value="" key="0" />
  {vehicleModels
    ? vehicleModels.map(otherEntity => (
        <option value={otherEntity.name} key={otherEntity.id}>
          {otherEntity.name}
        </option>
      ))
    : null}
</AvInput>
```

##### Correct code:
Change the name in the AvInput and the value in the option.

- Also, the link generated in the list view, does not have the URL format:

`<Link to={`vehicleModel/${vehicleVersion.vehicleModel.id}`}>{vehicleVersion.vehicleModel.name}</Link>`

When the correct link would use `vehicle-model`

##### **Motivation for or Use Case**

This generates a broken application, and should be easy to  fix for someone who knows the generator better than me, if not i will make sometime in the weekend to check if i cal help myself.

##### **Reproduce the error**

Create an application with the react generator, import the JDL in the description, create a brand and then try to create a model.
Also click on the link in the list of the model and a blank page is shown.

##### **Related issues**

I could not fin a related open bug

##### **Suggest a Fix**

1. Fix the view update generator to use the display property only in the html of the option
1. Fix the list view generator using the entity name with the URL format (string to minus with "-" instead camelCase

##### **Browsers and Operating System**

Ubuntu 18.04 LTS

- [X] Checking this box is mandatory (this is just to show you read everything)

**Comment by agaspardcilia** [comment]
I wasn't able to reproduce the error with the master. Can you make sure this is still an issue with the latest version of the code base? Thank you.

**Comment by agaspardcilia** [comment]
I think this bug was fixed with #7981.

**Comment by deepu105** [comment]
fixed
